sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -215,13 +215,18 @@ private object PostgresDialect extends JdbcDialect with SQLConfHelper {
       case sqlException: SQLException =>
         sqlException.getSQLState match {
           // https://www.postgresql.org/docs/14/errcodes-appendix.html
-          case "42P07" =>
+          case "42P07" if sqlException.getMessage != null =>
```

Comment:
If there's no message, what do we think that means? do one of the existing error paths still apply?

Reply:
Regexes in match expressions do not accept nulls. It's a defensive action. In general, there is no such case upstream. And even if it does occur, we will handle it properly in the default branch.